### PR TITLE
BL-1165-lc-call-number-sort

### PR DESCRIPTION
**Pull request by quoideneuf**
Processes lc call numbers into codes that will sort correctly in solr. 

Note that the real tests are in tul_cob/indices_spec.rb. Perhaps it would make sense at some point to add integration tests that use solr?

### CODE REVIEW

File: spec/cob_index/macros/custom_spec.rb [hunk 1]
```diff
@@ -2122,4 +2122,59 @@
...
+    before do
+      stub_const("ENV", ENV.to_hash.merge("SOLR_DISABLE_UPDATE_DATE_CHECK" => "false"))
+      subject.instance_eval do
+        to_field "lc_call_number_sort", extract_lc_call_number_sort
+        settings do
+          provide "marc_source.type", "xml"
+        end
+      end
+
+      ENV["SOLR_DISABLE_UPDATE_DATE_CHECK"] = nil
```
Comment (dkinzer) [inline spec/cob_index/macros/custom_spec.rb:2137]:
```suggestion
      subject.instance_eval do
        to_field "lc_call_number_sort", extract_lc_call_number_sort
        settings do
          provide "marc_source.type", "xml"
        end
      end
```

**Comment by dkinzer** [comment]
> Note that the real tests are in tul_cob/indices_spec.rb. Perhaps it would make sense at some point to add integration tests that use solr?

Yes, we're working on adding sol relevancy search tests to cob_index (which will require solr service running) so likely we could add some other tests too.

File: lib/cob_index/macros/custom.rb [hunk 2]
```diff
@@ -727,6 +727,58 @@ def extract_update_date
           acc << latest_date unless latest_date.empty?
         end
       end
+
+      def process_lc_call_sort(data, result = "")
+        @alphabet ||= ("a".."z").to_a
+
+        if data.is_a? Array
+          return data.map { |segment| process_lc_call_sort(segment) }.join("")
```
Comment (dkinzer) [inline lib/cob_index/macros/custom.rb:735]: Nice use of recursion here 😎

File: spec/cob_index/macros/custom_spec.rb (hunk 1 above)
Comment (quoideneuf) [reply spec/cob_index/macros/custom_spec.rb]: fixed - thanks

File: lib/cob_index/macros/custom.rb (hunk 2 above)
Comment (quoideneuf) [reply lib/cob_index/macros/custom.rb:735]: ♻️

**Pull request merged**